### ISSUE TITLE
DOCS: Stop requiring every commit to reference an issue

### ISSUE THREAD

**Issue by mvashishtha** [modin-project/modin#4982]
@jbrockmendel mentioned in #4885 that this requirement is painful. @vnlitvinov asked [here](https://github.com/modin-project/modin/issues/4747#issuecomment-1248027991) whether we should keep the requirement.

Similar projects pandas, vaex, dask, and polars do not seem to have this requirement.

I vote for relaxing this requirement but still [requiring](https://github.com/modin-project/modin/blob/0f70e828ed127229f7160075d2ad06a6b16b1002/commitlint.config.js#L8) commits to start with one of an allowed set of prefixes, currently `{"FEAT", "DOCS", "FIX", "REFACTOR", "TEST", "PERF"}`. The contributing guidelines should say that if the PR fixes an existing relevant issue, the commit message should include that issue's number.

@modin-project/modin-contributors @modin-project/modin-core please share your thoughts.

**Comment by pyrito** [comment]
Hmmm....I don't know how I feel about this. I liked the discipline of having associated issues with a PR. It decoupled the problem and the solution quite nicely. What I would definitely be onboard for is allowing REFACTOR, TEST, or DOCS PRs to not have an issues associated with them. I think PERF, FEAT, and FIX greatly benefit with having a commit refer to an issue.

**Comment by mvashishtha** [comment]
>  I think PERF, FEAT, and FIX greatly benefit with having a commit refer to an issue.

I think we could even drop FEAT here so we only require issues for PERF and FIX. For bug fixes it's really useful to have a reproducer (before, during, and after the PR) and requiring an issue will promote that. Same applies to PERF. FEAT issues tend to be simple and easier to verify, e.g. [all these](https://github.com/modin-project/modin/issues?q=is%3Aissue+is%3Aopen+%22function+implementation%22).

**Comment by pyrito** [comment]
@mvashishtha I've found that having discussions for FEAT in a separate issue usually helps bring a discussion and light some motivating examples (at least in my experience). I think we should keep issues for FEAT. Others are welcome to chime in.

**Comment by vnlitvinov** [comment]
What I would like to actually do is to _keep_ the same logic we were doing as regular contributors (i.e. creating issues for tracking purposes etc), but do not _require_ these issues to be created by external contributors.

In my opinion, accepting some random external contribution should be as frictionless as possible.

**Comment by mvashishtha** [comment]
In #5109, a new contributor is struggling to amend the first in a series of several commits to meet the commitlint requirement. Rebasing can be hard even for me, and I have a fair amount of git experience. I think we absolutely need to remove this requirement for new contributors, at least.

@anmyachev @vnlitvinov @pyrito @YarShev do you think it's ok to remove the commitlint requirement entirely? Or should we be more precise and require the commitlint for members of modin-project/modin-contributors?

**Comment by vnlitvinov** [comment]
I think the suggestion we talked offline of _moving_ the check from enforcing the specific first commit message pattern to enforcing specific PR title pattern feels much better.

Editing a PR title in Github is _waaaaaaaay_ easier than rebasing.

**Comment by RehanSD** [comment]
I agree with @vnlitvinov - that way we don't have to have a complicated set of rules for whose commits are formatted in a specific way for which people, but our commits (on master) are still nicely formatted!
